plotter/step.go

```diff
@@ -0,0 +1,173 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
```

Comment:
I have not considered transformation function because I think that step line is a quite common thing. It would be nice if I can use it out-of-box. Also it differs little bit in filling algorithm. We should not try to fill lonely vertical line. You probably would not nice difference anyway but it is just cleaner approach.

I have not considered `Line` enhancement because `Step` has additional data (`StepType` or better `StepKind`) that is useless in `Line`. However it can make a sense to make it a pointer or to add `NoStep` default constant and merge it with `Line`. We can reuse some code this way. It looks like a good idea for me. Let me reimplement it this way?

Also setting `ShadeColor` of 'Line' doesn't make its y-axis include 0 (zero) by default. I'm not sure if it is a bug or intentional. But it doesn't make a sense for `Step` because most bottom step wouldn't have a shade and it looks ugly. However this is the same for `Line` if bottom line is horizontal. So I think it is a bug in `Line`. And then there is no obstacles for merging them. This is fixed in `Step` already.

Reply:
And surely I'll update a date.